tests/framework/utils.py

```diff
@@ -206,3 +229,106 @@ def get_cpu_vendor():
     if 'AuthenticAMD' in brand_str:
         return CpuVendor.AMD
     return CpuVendor.INTEL
+
+
+def is_range(content):
+    """Return true if `content` is a range.
+
+    E.g ranges: 0-10.
+    """
+    match = re.search("([0-9][1-9]*)-([0-9][1-9]*)", content)
+    # Group is a singular value.
+    return match is not None
+
+
+def list_range(content):
+    """Return a range of integers based on the `content`.
+
+    The `content` respects the LIST FORMAT defined in the
+    cpuset documentation.
+    See: https://man7.org/linux/man-pages/man7/cpuset.7.html.
+    """
+    content = content.strip()
+    ends = content.split("-")
+    if len(ends) != 2:
+        return None
+
+    return list(range(int(ends[0]), int(ends[1]) + 1))
+
+
+def parse_list_format(content):
+    """Parse list formats for cpuset and mems.
+
+    See LIST FORMAT here: https://man7.org/linux/man-pages/man7/cpuset.7.html.
+    """
+    content = content.strip()
+    if len(content) == 0:
+        return []
+
+    groups = content.split(",")
+    arr = set()
+
+    def func(acc, cpu):
+        if is_range(cpu):
+            acc.update(list_range(cpu))
+        else:
+            acc.add(int(cpu))
+        return acc
+
+    return list(functools.reduce(func, groups, arr))
+
+
+def cpus_arr():
+    """Obtain the real processor map.
+
+    See this issue for details:
+    https://github.com/moby/moby/issues/20770.
+    """
+    cmd = "cat /proc/mounts | grep cgroup | grep cpuset | cut -d' ' -f2"
+    rc, stdout, stderr = run_cmd(cmd)
+    assert rc == 0
+    assert stderr == ""
+    cpuset_mountpoint = stdout.strip()
+
+    cmd = "cat {}/cpuset.cpus".format(cpuset_mountpoint)
+    rc, cpulist, stderr = run_cmd(cmd)
```

Comment:
what does `rc` stand for ?

Reply:
The return code is already accounted for inside the run_cmd function, [here](https://github.com/firecracker-microvm/firecracker/pull/2121/commits/00c3cc350e5024e3ebf7ab72ef4e9e917db9d4f9#diff-8c19ca5d8bafe38bb49b551cdc7a084bR149), so you could just run:

`_, cpulist, _ = run_cmd(cmd)`

Reply:
Done.